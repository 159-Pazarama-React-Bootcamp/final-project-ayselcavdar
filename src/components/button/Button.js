import styles from './Button.module.css';
import PropTypes from 'prop-types';

const Button = ({ disabled = false, content, ...props }) => {
  return (
    <button
      disabled={disabled}
      className={styles.btn}
      style={
        props.type === 'reset'
          ? {
              backgroundColor: '#E6E6E6',
              color: '#6B63FF',
            }
          : {}
      }
      {...props}
    >
      {content}
    </button>
  );
};

export default Button;

Button.propTypes = {
  disabled: PropTypes.bool,
  content: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};
