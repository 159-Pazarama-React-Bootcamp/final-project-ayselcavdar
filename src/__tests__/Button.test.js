import render from 'react-test-renderer';
import Button from '../components/button/Button';


describe('<Button /> rendering', () => {
  it('if button has disabled prop as true, it should be disabled', () => {
    const wrapper = shallow(
      <Button content={'test'} type="button" disabled={true} />
    );
    expect(wrapper.find('button').props().disabled).toBe(true);
  });

  it('Button\'s disabled prop change to false', () => {
    const wrapper = shallow(
      <Button content={'test'} type="button" disabled={false} />
    );
    expect(wrapper.find('button').props().disabled).toBe(false);
  });

  it('Button\'s content should be in button element', () => {
    const wrapper = shallow(
      <Button content={'test'} type="button" />
    );
    expect(wrapper.find('button').text()).toEqual('test');
  });

  it('Button snapshots check', () => {
    const comp = render.create(<Button content={'click'} type="button" />);
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
