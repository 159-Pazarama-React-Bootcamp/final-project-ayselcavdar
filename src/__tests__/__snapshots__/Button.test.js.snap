// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Button render check 1`] = `
<button
  className="btn"
  style={Object {}}
  type="button"
>
  click
</button>
`;

exports[`<Button /> rendering Button snapshots check 1`] = `
<button
  className="btn"
  style={Object {}}
  type="button"
>
  click
</button>
`;

exports[`Alert comp change className to props Button render check 1`] = `
<button
  className="btn"
  style={Object {}}
  type="button"
>
  click
</button>
`;

exports[`Button render check 1`] = `
<button
  className="btn"
  style={Object {}}
  type="button"
>
  click
</button>
`;
